pkg/k8s/api.go

```diff
@@ -22,9 +23,11 @@ import (
 var minAPIVersion = [3]int{1, 10, 0}
 
 // KubernetesAPI provides a client for accessing a Kubernetes cluster.
+// TODO: support spClient
```

Comment:
Is this something that has to happen in this PR?

If not, can this be more descriptive? What's `spClient`? Are there blockers? What's the benefit?

Reply:
`spClient` == Service Profile ClientSet. It doesn't have to happen in this PR, and maybe shouldn't, because I think we should move ServiceProfile code from `controller` into `pkg` (#2751).

I'll update the comment with all this context.